org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandlerTest.java

```diff
@@ -224,4 +231,65 @@ public void testDocumentFormattingWithCustomOption() throws Exception {
 		String newText = TextEditUtil.apply(unit, edits);
 		assertEquals(expectedText, newText);
 	}
+
+	@Test
+	public void testGoogleFormatter() throws Exception {
+		try {
+			String text =
+			//@formatter:off
+					"package org.sample;\n\n" +
+					"public class Baz {\n" +
+					"  String name;\n" +
+					"}\n";
+				//@formatter:on"
+			ICompilationUnit unit = getWorkingCopy("src/org/sample/Baz.java", text);
+			String uri = JDTUtils.toURI(unit);
+			TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+			FormattingOptions options = new FormattingOptions(2, true);// ident == 2 spaces
+			DocumentFormattingParams params = new DocumentFormattingParams(textDocument, options);
+			Bundle bundle = Platform.getBundle(JavaLanguageServerTestPlugin.PLUGIN_ID);
+			URL googleFormatter = bundle.getEntry("/testresources/eclipse-java-google-style.xml");
+			URL url = FileLocator.resolve(googleFormatter);
+			preferenceManager.getPreferences().setFormatterUrl(url.toExternalForm());
+			FormatterManager.configureFormatter(preferenceManager);
+			List<? extends TextEdit> edits = server.formatting(params).get();
+			assertNotNull(edits);
+			String newText = TextEditUtil.apply(unit, edits);
+			assertEquals(text, newText);
+		} finally {
+			preferenceManager.getPreferences().setFormatterUrl(null);
+			FormatterManager.configureFormatter(preferenceManager);
+		}
+	}
+
+	@Test
+	public void testGoogleFormatterFilePath() throws Exception {
+		try {
+			String text =
+			//@formatter:off
+					"package org.sample;\n\n" +
+					"public class Baz {\n" +
+					"  String name;\n" +
+					"}\n";
+				//@formatter:on"
+			ICompilationUnit unit = getWorkingCopy("src/org/sample/Baz.java", text);
+			String uri = JDTUtils.toURI(unit);
+			TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+			FormattingOptions options = new FormattingOptions(2, true);// ident == 2 spaces
+			DocumentFormattingParams params = new DocumentFormattingParams(textDocument, options);
+			Bundle bundle = Platform.getBundle(JavaLanguageServerTestPlugin.PLUGIN_ID);
+			URL googleFormatter = bundle.getEntry("/testresources/eclipse-java-google-style.xml");
+			URL url = FileLocator.resolve(googleFormatter);
+			File file = new File(url.toURI());
+			preferenceManager.getPreferences().setFormatterUrl(file.getAbsolutePath());
+			FormatterManager.configureFormatter(preferenceManager);
+			List<? extends TextEdit> edits = server.formatting(params).get();
+			assertNotNull(edits);
+			String newText = TextEditUtil.apply(unit, edits);
+			assertEquals(text, newText);
+		} finally {
+			preferenceManager.getPreferences().setFormatterUrl(null);
+			FormatterManager.configureFormatter(preferenceManager);
+		}
+	}
```

Comment:
can you also please add a test case for file paths containing spaces. We've been bitten so many times it's not even funny anymore